pkg/sites/loader.go

```diff
@@ -12,35 +13,61 @@ import (
 )
 
 // LoadComicFromSource will return a `comic` instance initialized based on the source
-func LoadComicFromSource(conf *config.ComicConfig, source, url, country, format string) (*core.Comic, error) {
+func LoadComicFromSource(conf *config.ComicConfig, source, url, country, format string) ([]core.Comic, error) {
 	var err error
+	var collection []core.Comic
+	var initializer func(*core.Comic) error
+	var issues []string
 
-	comic := &core.Comic{
-		Config:    conf,
-		URLSource: url,
-		Source:    source,
-		Format:    format,
-	}
+	options := map[string]string{"country": country}
 
 	switch source {
 	case "www.comicextra.com":
-		err = comicextra.Initialize(comic)
+		issues, err = comicextra.RetrieveIssueLinks(url)
+		initializer = comicextra.Initialize
 	case "mangarock.com":
-		if country != "" {
-			options := map[string]string{"country": country}
-			comic.Options = options
-		}
-		err = mangarock.Initialize(comic)
+		issues, err = mangarock.RetrieveIssueLinks(url, options)
+		initializer = mangarock.Initialize
 	case "www.mangareader.net":
-		err = mangareader.Initialize(comic)
+		issues, err = mangareader.RetrieveIssueLinks(url)
+		initializer = mangareader.Initialize
 	case "www.mangatown.com":
-		err = mangatown.Initialize(comic)
-	case "www.mangahere.cc":
-		err = fmt.Errorf("mangahere is currently disabled")
-	//sites.SetupMangaHere(comic)
+		issues, err = mangatown.RetrieveIssueLinks(url)
+		initializer = mangatown.Initialize
 	default:
 		err = fmt.Errorf("It was not possible to determine the source")
 	}
 
-	return comic, err
+	if err != nil {
+		return collection, err
+	}
+
+	collection, err = InitializeCollection(initializer, url, issues, conf, format, source, options)
+
+	return collection, err
+}
+
```

Comment:
Since this an exported function please add a comment explaining what it does

Reply:
Actually I think it shouldn't be exported.